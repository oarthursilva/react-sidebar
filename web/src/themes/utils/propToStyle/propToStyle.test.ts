import { propToStyle } from './index';

describe('propToStyle()', () => {
  describe('when propToStyle receives a sample argument', () => {
    test('and it is a string, should convert to style', () => {
      const propToStyleResult = propToStyle('textAlign');
      // <Text textAlign="center" />
      const componentProps = { textAlign: 'center' };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toEqual({ textAlign: 'center' });
    });

    test('and it is a number, should convert to style', () => {
      const propToStyleResult = propToStyle('flex');
      // <Box flex={1}" />
      const componentProps = { flex: 1 };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toEqual({ flex: 1 });
    });
  });

  describe('when propToStyle receives an argument with breakpoint', () => {
    test('render only a single small breakpoint resolution', () => {
      const propToStyleResult = propToStyle('textAlign');
      // <Text textAlign={{ xs: 'center' }} />
      const componentProps = { textAlign: { xs: 'center' } };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toMatchSnapshot();
    });

    test('render only a single extra large breakpoint resolution', () => {
      const propToStyleResult = propToStyle('textAlign');
      // <Text textAlign={{ lg: 'center' }} />
      const componentProps = { textAlign: { xl: 'center' } };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toMatchSnapshot();
    });
  });

  describe('when propToStyle receives an argument with breakpoints', () => {
    test('render multiple breakpoints resolution', () => {
      const propToStyleResult = propToStyle('textAlign');
      // <Text textAlign={{ sm: 'left', md: 'center', lg: 'right' }} />
      const componentProps = { textAlign: { sm: 'left', md: 'center', lg: 'right' } };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toMatchSnapshot();
    });
  });
});
