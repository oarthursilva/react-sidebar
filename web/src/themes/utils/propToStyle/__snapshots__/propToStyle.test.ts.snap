// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() when propToStyle receives an argument with breakpoint render only a single extra large breakpoint resolution 1`] = `
Array [
  "
        ",
  "
        @media only screen and (min-width: ",
  "1200",
  "px) {
          ",
  "text-align: center;",
  " ;
      }
      ",
  "
      ",
]
`;

exports[`propToStyle() when propToStyle receives an argument with breakpoint render only a single small breakpoint resolution 1`] = `
Array [
  "
        ",
  "
        @media only screen and (min-width: ",
  "0",
  "px) {
          ",
  "text-align: center;",
  " ;
      }
      ",
  "
      ",
]
`;

exports[`propToStyle() when propToStyle receives an argument with breakpoints render multiple breakpoints resolution 1`] = `
Array [
  "
        ",
  "
        @media only screen and (min-width: ",
  "480",
  "px) {
          ",
  "text-align: left;",
  " ;
      }
      ",
  "
        @media only screen and (min-width: ",
  "768",
  "px) {
          ",
  "text-align: center;",
  " ;
      }
      ",
  "
        @media only screen and (min-width: ",
  "992",
  "px) {
          ",
  "text-align: right;",
  " ;
      }
      ",
  "
      ",
]
`;
